controllers/chat/group/changeChannelDetail.js

```diff
@@ -27,6 +31,42 @@ const changeChannelDetail = async (req, res) => {
   if (channel_name) updateData.name = channel_name;
 
   try {
+    const members = channel.data?.better_channel_member?.map((member) => member.user_id);
+
+    if (channel_name) {
+      const textOwnUser = `You changed the group name to "${channel_name}"`;
+      const textTargetUser = `${ownUser.username} changed the group name to "${channel_name}"`;
+      const textDefaultUser = `${ownUser.username} changed the group name to "${channel_name}"`;
+
+      await channel.sendMessage({
+        text: textDefaultUser,
+        own_text: textOwnUser,
+        other_text: textTargetUser,
+        type: 'system',
+        user_id: userId,
+        system_user: userId,
+        isSystem: true,
+        members: members
+      });
+    }
+
+    if (channel_image) {
+      const textOwnUser = `You changed the group image`;
+      const textTargetUser = `${ownUser.username} changed the group image`;
+      const textOtherUser = `${ownUser.username} changed the group image`;
+
+      await channel.sendMessage({
+        text: textOtherUser,
+        own_text: textOwnUser,
+        other_text: textTargetUser,
+        type: 'system',
+        user_id: userId,
+        system_user: userId,
+        isSystem: true,
+        members: members
+      });
+    }
+
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

This block of code seems to be repeating itself with minor changes. Consider creating a helper function to reduce redundancy and improve maintainability. Also, there's a repeated variable `textDefaultUser` and `textOtherUser` which have the same value. You can remove one of them.
```diff
-      const textOwnUser = `You changed the group name to "${channel_name}"`;
-      const textTargetUser = `${ownUser.username} changed the group name to "${channel_name}"`;
-      const textDefaultUser = `${ownUser.username} changed the group name to "${channel_name}"`;
-
-      await channel.sendMessage({
-        text: textDefaultUser,
-        own_text: textOwnUser,
-        other_text: textTargetUser,
-        type: 'system',
-        user_id: userId,
-        system_user: userId,
-        isSystem: true,
-        members: members
-      });
-
-      const textOwnUser = `You changed the group image`;
-      const textTargetUser = `${ownUser.username} changed the group image`;
-      const textOtherUser = `${ownUser.username} changed the group image`;
-
-      await channel.sendMessage({
-        text: textOtherUser,
-        own_text: textOwnUser,
-        other_text: textTargetUser,
-        type: 'system',
-        user_id: userId,
-        system_user: userId,
-        isSystem: true,
-        members: members
-      });

+      const sendSystemMessage = async (action, target) => {
+        const textOwnUser = `You ${action} the group ${target}`;
+        const textTargetUser = `${ownUser.username} ${action} the group ${target}`;
+
+        await channel.sendMessage({
+          text: textTargetUser,
+          own_text: textOwnUser,
+          other_text: textTargetUser,
+          type: 'system',
+          user_id: userId,
+          system_user: userId,
+          isSystem: true,
+          members: members
+        });
+      }
+
+      if (channel_name) {
+        await sendSystemMessage('changed name to', `"${channel_name}"`);
+      }
+
+      if (channel_image) {
+        await sendSystemMessage('changed image', '');
+      }
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->